src/zenml/zen_stores/sql_zen_store.py

```diff
@@ -592,25 +576,24 @@ def _save_stack(
                     session.delete(result)
 
             for ctype, cname in stack_configuration.items():
-                statement = (
```

Comment:
This was never needed, because the `ZenStackDefinition.stack_name <-> ZenStack.name` foreign key constraint prevents it, and we delete the definitions above, if the stack exists anyway.